server/src/main/java/io/crate/geo/GeoJSONUtils.java

```diff
@@ -131,9 +132,14 @@ public static Map<String, Object> shape2Map(Shape shape) {
         }
     }
 
+    /**
+     * Transforms Geometry Collection to MultiPoint or MultiPolygon
+     * if all shapes in collection are points/polygons.
+     */
```

Comment:
This relates to the how `JtsSpatial` parses wtk strings. 
```suggestion
    /**
     * Convert a WKT string into a Shape.
     *
     * GeometryCollections composed of a single part are converted
     * into the respective multi-part (MultiPoint, MultiLineString, or MultiPolygon) objects
     * to maximize interoperability.
     * See {@link https://datatracker.ietf.org/doc/html/draft-ietf-geojson-03#section-3.1.8}
     */
```